src/java/org/apache/cassandra/service/accord/AccordService.java

```diff
@@ -341,16 +357,17 @@ public IVerbHandler<? extends Request> verbHandler()
         return requestHandler;
     }
 
-    @Override
-    public long barrier(@Nonnull Seekables keysOrRanges, long epoch, long queryStartNanos, long timeoutNanos, BarrierType barrierType, boolean isForWrite)
+    private <S extends Seekables<?, ?>> long barrier(@Nonnull S keysOrRanges, long epoch, long queryStartNanos, long timeoutNanos, BarrierType barrierType, boolean isForWrite, BiFunction<Node, S, AsyncResult<SyncPoint<S>>> syncPoint)
     {
         AccordClientRequestMetrics metrics = isForWrite ? accordWriteMetrics : accordReadMetrics;
         TxnId txnId = null;
         try
         {
             logger.debug("Starting barrier key: {} epoch: {} barrierType: {} isForWrite {}", keysOrRanges, epoch, barrierType, isForWrite);
             txnId = node.nextTxnId(Kind.SyncPoint, keysOrRanges.domain());
-            AsyncResult<Timestamp> asyncResult = node.barrier(keysOrRanges, epoch, barrierType);
+            AsyncResult<Timestamp> asyncResult = syncPoint == null
+                                                 ? node.barrier(keysOrRanges, epoch, barrierType)
+                                                 : Barrier.barrier(node, keysOrRanges, epoch, barrierType, syncPoint);
```

Comment:
One version launches straight into `Barrier.barrier` the other goes through `node.barrier`. Should we pick a single level of interfacing with Accord here?